### PR TITLE
yukon: battery/charger: fixes and cleanup

### PR DESCRIPTION
**Pull request by Dudulle**

### CODE REVIEW

**Comment by Dudulle** [comment]
The last commit requires https://github.com/sonyxperiadev/device-sony-flamingo/pull/43 to work as intended on flamingo.

File: arch/arm/boot/dts/qcom/batterydata-yukon-flamingo.dtsi [hunk 1]
```diff
@@ -72,7 +72,7 @@ qcom,flamingo-batterydata {
 	};
 
 	qcom,rbatt-sf-lut {
-		qcom,lut-col-legend = <(-20) 0 25 40 60 70>;
```
Comment (kholk) [inline arch/arm/boot/dts/qcom/batterydata-yukon-flamingo.dtsi:75]: This looks REALLY wrong. You are lowering the algorithm precision by removing that extra value.

Comment (Dudulle) [inline arch/arm/boot/dts/qcom/batterydata-yukon-flamingo.dtsi:75]: I'm pretty sure this value was added by mistake. I checked in kernel 18.5.C.0.25 and it is not there.
Besides that, if the value is correct, then there would be some data missing in qcom,lut-data.

Comment (kholk) [inline arch/arm/boot/dts/qcom/batterydata-yukon-flamingo.dtsi:75]: My bad, then :)

**Pull request merged**